File: mall-chat-web/src/api/category.ts
import { BASE_URL, request } from "./http";

export type CategoryItem = {
    id: number;
    name: string;
    parentId?: number | null;
    level: number;
    sort: number;
};

export function fetchCategories() {
    return request<CategoryItem[]>(BASE_URL, "/categories");
}

File: mall-chat-service/src/main/java/com/mallchat/model/Category.java
package com.mallchat.model;

import java.time.LocalDateTime;
import lombok.Data;

/**
 * Post category entity.
 */
@Data
public class Category {
    private Long id;
    private String name;
    private Long parentId;
    private Integer level;
    private Integer sort;
    private Integer status;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
